validator/src/main/java/com/amazon/aoc/validators/TraceValidator.java

```diff
@@ -88,61 +88,56 @@ public void validate() throws Exception {
 
   // this method will hit get trace from x-ray service and get retrieved trace
   private Map<String, Object> getRetrievedTrace(List<String> traceIdList) throws Exception {
-    AtomicReference<List<Trace>> retrieveTraceListAtomicReference = new AtomicReference<>();
+
+    AtomicReference<Map<String, Object>> flattenedJsonMapForRetrievedTrace = null;
     RetryHelper.retry(
-        3,
+        5,
         () -> {
           List<Trace> retrieveTraceList = null;
           retrieveTraceList = xrayService.listTraceByIds(traceIdList);
-          retrieveTraceListAtomicReference.set(retrieveTraceList);
-
           if (retrieveTraceList == null || retrieveTraceList.isEmpty()) {
             throw new BaseException(ExceptionCode.EMPTY_LIST);
           }
+
+          // in case the json format is wrong, retry it.
```

Comment:
What do you mean by json format is wrong?

Reply:
check this: https://github.com/aws-observability/aws-otel-collector/runs/1397013128?check_suite_focus=true, the "e2etest-ecs (otlp_trace, EC2)" one, looks like xray returned a empty/invalid json of segment. but i can't reproduce it on my local even with the same trace id, so i suspect xray backend somehow returned a invalid json